core/primitives/src/views.rs

```diff
@@ -1660,6 +1660,36 @@ impl ExecutionOutcomeWithIdView {
     }
 }
 
+pub struct TxStatusView {
+    pub execution_outcome: Option<FinalExecutionOutcomeViewEnum>,
+    pub status: TxExecutionStatus,
+}
+
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    Clone,
+    Debug,
+    Eq,
+    PartialEq,
+)]
+pub enum TxExecutionStatus {
```

Comment:
Thinking about the corner cases from a point of view of a user:
* Submit a transaction
* Transaction gets included in a block in a fork
* See that the transaction is included in a block
* Wait for 10s
* See that the transaction is included in a block
* Wait for 10s
* .. repeat forever

Should we help the user in this case?
IIUC, this infinite loop is not possible with the execution of receipts, because the outgoing receipts of the parent block **must be** included in the child block.
Inclusion of transactions is different, because recording a transaction on-chain is not a mandatory action.

A counter-argument would be that a transaction may never be included in any block by chunk producers, and we can't possibly verify this condition.

Reply:
After we merge this, I'll start working on adding the same struct to the request to all the methods touched in this PR. It should solve this issue at least partially.

Thank you for the review!
I've addressed all the other comments and added the notes to CHANGELOG.md, please have a look again